src/Inkluzitron/Modules/InviteModule.cs

```diff
@@ -0,0 +1,119 @@
+
+        private BotDatabaseContext DbContext { get; set; }
+        private DatabaseFactory DatabaseFactory { get; }
+        private UsersService UsersService { get; }
+
+        public InviteModule(IConfiguration config,
+            DatabaseFactory databaseFactory,
+            UsersService usersService,
+            DiscordSocketClient discordSocketClient)
+        {
+            Config = config;
+            DatabaseFactory = databaseFactory;
+            UsersService = usersService;
+            DiscordSocketClient = discordSocketClient;
+        }
+
+        override protected void BeforeExecute(CommandInfo command)
+        {
+            DbContext = DatabaseFactory.Create();
+            base.BeforeExecute(command);
+        }
+
+        override protected void AfterExecute(CommandInfo command)
+        {
+            DbContext?.Dispose();
+            base.AfterExecute(command);
+        }
+
+        [Command("")]
+        [Summary("Vytvoří jednorázový invite link a pošle ho do DM.")]
+        [RequireBotPermission(GuildPermission.CreateInstantInvite)]
+        public async Task CreateInviteAsync()
+        {
+            var invite = await Context.Guild.DefaultChannel.CreateInviteAsync(
+                null,
+                2, // create 2 uses for invite because if only 1 usage is available it's automatically deleted after joining
+                false,
+                true,
+                new RequestOptions()
+                {
+                    AuditLogReason = "User with ID " +
+                                     Context.User.Id +
+                                     " and name " +
+                                     Context.User.Username +
+                                     " wants to invite someone!"
+                });
+
+            var user = await DbContext.GetOrCreateUserEntityAsync(
+                Context.Message.Author);
+
+            var inviteDb = new Invite
+            {
+                GeneratedAt = DateTime.Now,
+                GeneratedByUserId = user.Id,
+                InviteLink = invite.Url,
+            };
+
+            await DbContext.Invites.AddAsync(inviteDb);
+            await DbContext.SaveChangesAsync();
+
+            await Context.User.SendMessageAsync(
+                "Byl ti vygenerován následující invite link: " + invite.Url);
+            await ReplyAsync(
+                "Do DM jsem ti poslal vygenerovaný invite link. " + Config["MCrack"]);
+        }
+
+        [Command("blame")]
+        [Summary("Zjistí kým byl odesílatel nebo zadaný uživatel pozván.")]
+        public async Task BlameInviterAsync([Name("kdo")] IUser target = null)
+        {
+            target ??= Context.User;
+
+            var invites = DbContext.Invites
+                .Include(i => i.GeneratedBy)
+                .Include(i => i.UsedBy).AsQueryable();
+
+            foreach (var invite in invites)
+            {
+                if (invite.UsedByUserId != target.Id) continue;
```

Comment:
Tohle bych dal jako dotaz přímo do DB a ne vytahovat si celou tabulku a iterovat to až v appce. :notlikethis: